### PR TITLE
Adding test to electron system browser sample

### PR DESCRIPTION
**Pull request by salman90**
Added e2e tests to electron system browser sample to test acquireTokenInteractive API.

### CODE REVIEW

File: samples/e2eTestUtils/ElectronPlaywrightTestUtils.ts [hunk 1]
```diff
@@ -112,3 +112,16 @@ export function createFolder(foldername: string) {
         fs.mkdirSync(foldername, { recursive: true });
     }
 }
+
+export async function getAuthCodeUrl(page: Page): Promise<string> {
+    const msgPromise = await page.waitForEvent("console", {
+        predicate: async (message) => {
+            const text = message.text();
+            if (text.includes("https://login.microsoftonline.com")) {
```
Comment (peterzenz) [inline samples/e2eTestUtils/ElectronPlaywrightTestUtils.ts]: Better would be the text starts with the string "https://login.microsoftonline.com/"  Adding the trailing slash is a good practice as it prevents domains from matching that look like https://login.microsoftonline.com.attackerdomain.com/"

**Review by peterzenz — APPROVED** [review]

File: samples/msal-node-samples/ElectronSystemBrowserTestApp/src/Main.ts [hunk 2]
```diff
@@ -104,6 +104,10 @@ export default class Main {
         Main.fetchManager = new FetchManager();
         Main.registerSubscriptions();
         Main.attemptSSOSilent();
+
+        if (process.env.automation != null) {
```
Comment (derisen) [inline samples/msal-node-samples/ElectronSystemBrowserTestApp/src/Main.ts]: can we check for the actual expected value of this instead (here and elsewhere)?

File: samples/msal-node-samples/ElectronSystemBrowserTestApp/src/Main.ts [hunk 3]
```diff
@@ -104,6 +104,10 @@ export default class Main {
...
+            Main.authProvider.publishTest = Main.publish;
```
Comment (derisen) [inline samples/msal-node-samples/ElectronSystemBrowserTestApp/src/Main.ts]: nit: `publishForTest`

File: samples/msal-node-samples/ElectronSystemBrowserTestApp/tests/electron-code-aad.spec.ts [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+import {
+    SCREENSHOT_BASE_FOLDER_NAME,
+    validateCacheLocation,
+} from "../../testUtils";
+
+import * as path from "path";
+
+let electronApp: ElectronApplication;
+let page: Page;
+let browser: any;
```
Comment (derisen) [inline samples/msal-node-samples/ElectronSystemBrowserTestApp/tests/electron-code-aad.spec.ts]: type for the browser?

File: samples/msal-node-samples/ElectronSystemBrowserTestApp/tests/electron-code-aad.spec.ts [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+                username,
+                accountPwd
+            );            
+            const cachedTokens = await NodeCacheTestUtils.waitForTokens(
+                TEST_CACHE_LOCATION,
+                2000
+            );
+            expect(cachedTokens.accessTokens.length).toBe(1);
+            expect(cachedTokens.idTokens.length).toBe(1);
+            expect(cachedTokens.refreshTokens.length).toBe(1);
```
Comment (derisen) [inline samples/msal-node-samples/ElectronSystemBrowserTestApp/tests/electron-code-aad.spec.ts]: Can we add another expect clause here to show that the UI updates as expected?

File: samples/msal-node-samples/ElectronSystemBrowserTestApp/tests/electron-code-aad.spec.ts [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    });
+
+    test.afterEach(async () => {
+        await page.close();
+        await browserPage.close();
+        await NodeCacheTestUtils.resetCache(TEST_CACHE_LOCATION);
+    });
+
+    test("Acquire token by auth code", async () => {
+        try {
```
Comment (derisen) [inline samples/msal-node-samples/ElectronSystemBrowserTestApp/tests/electron-code-aad.spec.ts]: I think there is no use for the try catch block here

File: .github/workflows/msal-node-e2e.yml [hunk 7]
```diff
@@ -209,9 +209,13 @@ jobs:
         os: [windows-latest, macos-latest]
         sample:
           - 'ElectronTestApp'
+          - 'ElectronSystemBrowserTestApp'
 
     name: ${{ matrix.sample }} - ${{ matrix.os }}
 
+    env:
+        PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD: 1
```
Comment (derisen) [inline .github/workflows/msal-node-e2e.yml]: do we really need these env vars outside of the sample working directory?

File: samples/msal-node-samples/ElectronSystemBrowserTestApp/playwright.config.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+        trace: "on-first-retry",
+    },
+    projects: [
+        {
+            name: "chromium",
+            use: { ...devices["Desktop Chrome"] },
+        },
+    ],
+    timeout: 30000,
+    globalTimeout: 5400000,
```
Comment (derisen) [inline samples/msal-node-samples/ElectronSystemBrowserTestApp/playwright.config.ts:19]: can we add `retries` to the config, using `RETRY_TIMES` constant from e2eTestUtils? (I think testUtils.ts)

File: samples/e2eTestUtils/ElectronPlaywrightTestUtils.ts [hunk 9]
```diff
@@ -112,3 +112,16 @@ export function createFolder(foldername: string) {
         fs.mkdirSync(foldername, { recursive: true });
     }
 }
+
+export async function getAuthCodeUrl(page: Page): Promise<string> {
```
Comment (derisen) [inline samples/e2eTestUtils/ElectronPlaywrightTestUtils.ts]: let's rename this function. something like "retrieveAuthCodeUrlFromBrowserContext" would be better at signaling what it does

File: samples/msal-node-samples/ElectronSystemBrowserTestApp/tests/electron-code-aad.spec.ts (hunk 4 above)
Comment (salman90) [reply samples/msal-node-samples/ElectronSystemBrowserTestApp/tests/electron-code-aad.spec.ts]: Done!

File: samples/msal-node-samples/ElectronSystemBrowserTestApp/tests/electron-code-aad.spec.ts (hunk 5 above)
Comment (salman90) [reply samples/msal-node-samples/ElectronSystemBrowserTestApp/tests/electron-code-aad.spec.ts]: Updated!

File: samples/msal-node-samples/ElectronSystemBrowserTestApp/tests/electron-code-aad.spec.ts (hunk 6 above)
Comment (salman90) [reply samples/msal-node-samples/ElectronSystemBrowserTestApp/tests/electron-code-aad.spec.ts]: Removed try & catch

File: samples/msal-node-samples/ElectronSystemBrowserTestApp/playwright.config.ts (hunk 8 above)
Comment (salman90) [reply samples/msal-node-samples/ElectronSystemBrowserTestApp/playwright.config.ts:19]: Done

File: samples/e2eTestUtils/ElectronPlaywrightTestUtils.ts (hunk 1 above)
Comment (salman90) [reply samples/e2eTestUtils/ElectronPlaywrightTestUtils.ts]: Done

File: samples/e2eTestUtils/ElectronPlaywrightTestUtils.ts (hunk 9 above)
Comment (salman90) [reply samples/e2eTestUtils/ElectronPlaywrightTestUtils.ts]: done

File: samples/msal-node-samples/ElectronSystemBrowserTestApp/src/Main.ts (hunk 3 above)
Comment (salman90) [reply samples/msal-node-samples/ElectronSystemBrowserTestApp/src/Main.ts]: Done

File: samples/msal-node-samples/ElectronSystemBrowserTestApp/src/Main.ts (hunk 2 above)
Comment (salman90) [reply samples/msal-node-samples/ElectronSystemBrowserTestApp/src/Main.ts]: done

File: .github/workflows/msal-node-e2e.yml (hunk 7 above)
Comment (salman90) [reply .github/workflows/msal-node-e2e.yml]: Passing the env variables on package installation was breaking the tests. I need to declare the env variables before the package installation, so I moved the location of the env variables to be declared before all steps. Let me know if you have a better suggestion, but according to the [documentation](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idenv), it should be fine.

**Review by derisen — APPROVED** [review]
LGTM, thanks!

File: .github/workflows/msal-node-e2e.yml (hunk 7 above)
Comment (salman90) [reply .github/workflows/msal-node-e2e.yml]: @derisen it works fine when moving the env variables to the `Install Test Tools` step. Thanks for the feedback!

**Pull request merged**